Model/Resolver/GAResolver.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Elgentos\ServerSideAnalytics\Model\Resolver;
+
+use Exception;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlAuthorizationException;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Quote\Api\CartRepositoryInterface;
+use Magento\Quote\Model\MaskedQuoteIdToQuoteIdInterface;
+
+class GAResolver implements ResolverInterface
+{
+    /**
+     * @var CartRepositoryInterface
+     */
+    protected $quoteRepository;
+
+    /**
+     * @var MaskedQuoteIdToQuoteIdInterface
+     */
+    protected $maskedQuoteIdToQuoteId;
+
+    /**
+     * Construct.
+     * @param CartRepositoryInterface $quoteFactory
+     * @param MaskedQuoteIdToQuoteIdInterface $maskedQuoteIdToQuoteId
+     */
+    public function __construct(
+        CartRepositoryInterface $quoteRepository,
+        MaskedQuoteIdToQuoteIdInterface $maskedQuoteIdToQuoteId
+    ) {
+        $this->quoteRepository = $quoteRepository;
+        $this->maskedQuoteIdToQuoteId = $maskedQuoteIdToQuoteId;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        if (is_numeric($args['cartId']) && false === $context->getExtensionAttributes()->getIsCustomer()) {
```

Comment:
Please ensure you give a fallback/exception for `$args['cartId']` in case it doesn't exist.

Reply:
The cartId is the only place where this is not needed, this is because it is already made required by the `!` after the `String!` in the definition.

If you forget the cart id you will get the following response
![image](https://github.com/elgentos/magento2-serversideanalytics/assets/15870933/d3a75639-e481-4042-acc5-3a089f87503f)